server.js

```diff
@@ -18,33 +18,25 @@ app.use("/public", express.static(__dirname + "/public"));
 app.use(cookieParser());
 
 app.use((req, res, next) => {
-    console.log(app.locals.session_id);
+const { users } = require('./users')
+    console.log(users);
     const { cookies } = req
     try {
         if (cookies.session_id === undefined) {
             siteControl.authenticated = false;
-            console.log("please login");
             next()
         } else {
-            if (cookies.session_id == app.locals.session_id) {
+            if (sessions.hasOwnProperty(cookies.session_id)) {
```

Comment:
While `.hasOwnProperty` is perfectly functional and working, it's also a bit old-fashioned. Nowadays you'd just do something like

```suggestion
            if (sessions[cookies.session_id]) {
```

Especially since you know `sessions[cookies.session_id]` will be a non-empty string if it exists.